frontend/src/metabase-lib/queries/utils/index.js

```diff
@@ -70,7 +70,7 @@ export function cleanQuery(query) {
     _.all(filter, a => a != null),
   );
   if (filters.length > 0) {
-    query.filter = ["and", ...filters];
+    query.filter = QUERY.getFilterClause(filters);
```

Comment:
nit: if this was TypeScript, we'd get an error here: `QUERY.getFilterClause` can return `undefined`, but `query.filter` must be an array (or be absent from the object).
This case is impossible due to `if (filters.length > 0)` though.

Reply:
good catch, I met such situations several times, but it's going to be a problem of someone in the future when he decided to switch those files to TS 😅